proposals/pubsub.md

```diff
@@ -0,0 +1,182 @@
+-->
+
+- PubSub becomes more reliable with more nodes on the network.
+- Bulding (near) real-time products becomes easier, IPFS nodes are suitable out of the box.
+
+#### Leverage
+_How much would nailing this project improve our knowledge and ability to execute future projects?_
+
+<!--
+Explain the opportunity or leverage point for our subsequent velocity/impact (e.g. by speeding up development, enabling more contributors, etc)
+-->
+
+- Improves velocity of teams building products requiring pubsub by removing additional burden.
+- Improves pubsub reliability by adding more enabled nodes to the network.
+- Enables development of new features that require pubsub.
+
+#### Confidence
+_How sure are we that this impact would be realized? Label from [this scale](https://medium.com/@nimay/inside-product-introduction-to-feature-priority-using-ice-impact-confidence-ease-and-gist-5180434e5b15)_.
+
+<!--Explain why this rating-->
+
+Level 3
+
+- We know teams building products that depend pubsub.
+- Features like IPNS over pubsub depends on this.
+
+
+## Project definition
+#### Brief plan of attack
+
+<!--Briefly describe the milestones/steps/work needed for this project-->
+
+- Measure what is the overhead on nodes that do not directly use pubsub
+- If overhead is neglegable ship a release of go-ipfs with pubsub enabled (see [ipfs/go-ipfs#6621][])
+- Enable / upgrade IPFS nodes operated by PL.
+- Raise awareness (blog, tweet, support community transition)
+
+#### What does done look like?
+_What specific deliverables should completed to consider this project done?_
+
+- Shipped release of go-ipfs with pubsub enabled
+- Increased number of pubsub enabled IPFS nodes in the network.
+
+####  What does success look like?
+_Success means impact. How will we know we did the right thing?_
+
+- Various projects (like OrbitDB, [Forest](https://github.com/forestpm/forest)) no longer have to explain with instructions how to enable pubsub.
+- Teams like fission deliver pubsub to their users requesting it.
+- Pubsub becomes more reliable due to increased number of enabled nodes. 
+
+
+<!--
+Provide success criteria. These might include particular metrics, desired changes in the types of bug reports being filed, desired changes in qualitative user feedback (measured via surveys, etc), etc.
+-->
+
+#### Counterpoints &amp; pre-mortem
+_Why might this project be lower impact than expected? How could this project fail to complete, or fail to be successful?_
+
+- Everyone enables pubsub already anyway.
+- Pusub prooves (more) unreliable even with increased number of enabled nodes.
+
+#### Alternatives
+_How might this project’s intent be realized in other ways (other than this project proposal)? What other potential solutions can address the same need ?_
+
+- Alternative messaging primitives (e.g. Federated SSB like pubs)
+- [SMTP][] support in IPFS nodes (you can email any node by email address)
```

Comment:
Is that actually an alternative?  Will it give the "realtime" that Fission and others are hoping for?

Reply:
I use https://delta.chat/ as a realtime chat application that uses email protocol. In my experience it seems to work just fine, leading me  to believe that it could. That said going that route would most definitely require closer analyzes.

@dignifiedquire who contributed to that project might have less anecdotal evidence to support or disprove this.

Reply:
The performance/throughput/guarantees would be vastly different. Our pubsub protocols are useful for low-bandwidth, low-latency, real-time, unreliable broadcast. Email is good for point-to-point (broadcast gets expensive), reliable events.